src/Bicep.Types/Index/TypeSettings.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Azure.Bicep.Types.Index
+{
+    public class TypeSettings
+    {
+        [JsonConstructor]
+        public TypeSettings(string name, string version, bool isSingleton, TypeLocation configuration)
+            => (Name, Version, IsSingleton, Configuration) = (name, version, isSingleton, configuration);
+
+        public string Name { get; }
+
+        public string Version { get; }
+
+        public bool IsSingleton { get; }
+
+        public TypeLocation? Configuration { get; }
```

Comment:
[nit]: Should this maybe be called `ConfigurationType` or something similar? Just based on the name, I assumed this was configuration for the TypeIndex itself, but since it's a `TypeLocation` this must be the type against which to validate supplied configuration, right?

Reply:
Had it set to auto merge, but I agree with this suggestion? @anthony-c-martin thoughts?

Reply:
Sure - happy to approve another PR to change this!